adapters/pixfuture/pixfuture.go

```diff
@@ -0,0 +1,104 @@
+package pixfuture
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Pixfuture adapter.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpoint: config.Endpoint,
+	}, nil
+}
+
+// MakeRequests prepares and serializes HTTP requests to be sent to the Pixfuture endpoint.
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{Message: "No impressions in the bid request"}}
+	}
```

Comment:
This check is not required here as this is being checked in the core codebase already. Here is the link - https://github.com/prebid/prebid-server/blob/32fdbc4be35375912ac886ca05f7318c06995e0c/endpoints/openrtb2/auction.go#L770-L772